src/fondant/cli.py

```diff
@@ -184,72 +170,85 @@ def register_compile(parent_parser):
         formatter_class=argparse.RawDescriptionHelpFormatter,
         description=textwrap.dedent(
             """
-        Compile a fondant pipeline into either a docker-compose.yml(local) or kubeflow spec file.
+        Compile a fondant pipeline into pipeline specification file file.
 
         The pipeline argument is a formatstring. The compiler will try to import the pipeline from the module specified in the formatstring.
         (NOTE: path is patched to include the current working directory so you can do relative imports)
 
-        The --local or --kubeflow flag specifies the mode in which the pipeline will be compiled.
-        You can use the --extra-volumes flag to specify extra volumes to mount in the containers this can be used:
-
-        - to mount data directories to be used by the pipeline (note that if your pipeline's base_path is local it will already be mounted for you).
-        - to mount cloud credentials (see examples))
+        You can use different modes for fondant runners. Current existing modes are local and kubeflow.
 
-        Example:
-        fondant compile my_project.my_pipeline.py --local --extra-volumes $HOME/.aws/credentials:/root/.aws/credentials
+        Examples of compiling component:
+        fondant compile local --extra-volumes $HOME/.aws/credentials:/root/.aws/credentials my_project.my_pipeline.py
 
-        fondant compile my_project.my_pipeline.py --kubeflow --extra-volumes $HOME/.config/gcloud/application_default_credentials.json:/root/.config/gcloud/application_default_credentials.json
+        fondant compile kubeflow --extra-volumes $HOME/.config/gcloud/application_default_credentials.json:/root/.config/gcloud/application_default_credentials.json my_project.my_pipeline.py
         """,
         ),
     )
+
+    compiler_subparser = parser.add_subparsers()
+
     parser.add_argument(
         "ref",
         help="""Reference to the pipeline to run, can be a path to a spec file or
             a module containing the pipeline instance that will be compiled first (e.g. pipeline.py)
             """,
         action="store",
     )
-    # add a mutually exclusive group for the mode
-    mode_group = parser.add_mutually_exclusive_group(required=True)
-    mode_group.add_argument("--local", action="store_true")
-    mode_group.add_argument("--kubeflow", action="store_true")
 
-    parser.add_argument(
+    local_parser = compiler_subparser.add_parser(name="local", help="Local compiler")
+    kubeflow_parser = compiler_subparser.add_parser(
+        name="kubeflow",
+        help="Kubeflow compiler",
+    )
+
+    # Local runner parser
+    local_parser.add_argument(
         "--output-path",
         "-o",
-        help="Output directory",
-        default=None,
+        help="Output path of compiled pipeline",
+        default="docker_compose.yml",
```

Comment:
Is there a reason for going away from the default docker-compose naming? 

```suggestion
        default="docker-compose.yml",
```

Reply:
Good catch, forgot that it was referenced like this